File: backend_php/api/points.php
<?php
// backend/api/points.php
include_once '../config/cors.php';
include_once '../config/database.php';
include_once '../utils/JWT.php';

$database = new Database();
$db = $database->getConnection();

// Auth Check
$headers = getallheaders();
$jwt = isset($headers['Authorization']) ? str_replace('Bearer ', '', $headers['Authorization']) : "";
$user_data = JWT::decode($jwt);

if (!$user_data || $user_data['data']['role'] !== 'hod') {
    http_response_code(401);
    echo json_encode(["message" => "Unauthorized. HOD access required."]);
    exit();
}

// Handle GET (Fetch configs) or POST (Update configs)
if ($_SERVER['REQUEST_METHOD'] === 'GET') {
    // In a real DB, fetch from points_config table. 
    // Mocking response for now as per minimal viable setup.
    echo json_encode([
        ["id" => 1, "type" => "Certificate (Course)", "points" => 50],
        ["id" => 2, "type" => "Certificate (Internship)", "points" => 100],
        ["id" => 3, "type" => "Project (Hackathon)", "points" => 150],
        ["id" => 4, "type" => "Research Paper", "points" => 200]
    ]);
} elseif ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $data = json_decode(file_get_contents("php://input"));
    // Update logic here...
    echo json_encode(["message" => "Points configuration updated successfully"]);
}
?>

File: backend/app/routers/points.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.database import get_db
from app import auth, models, schemas

router = APIRouter(tags=["Points"])

@router.get("/points-config")
def get_points_config(
    current_user: models.User = Depends(auth.get_current_hod_user),
    db: Session = Depends(get_db)
):
    # Mock data for now as per minimal setup
    return [
        {"id": 1, "type": "Certificate (Course)", "points": 50},
        {"id": 2, "type": "Certificate (Internship)", "points": 100},
        {"id": 3, "type": "Project (Hackathon)", "points": 150},
        {"id": 4, "type": "Research Paper", "points": 200}
    ]

@router.post("/points-config")
def update_points_config(
    data: schemas.PointConfigBase,
    current_user: models.User = Depends(auth.get_current_hod_user),
    db: Session = Depends(get_db)
):
    # Logic to update
    return {"message": "Config updated"}
